virttest/virsh.py

```diff
@@ -621,6 +624,93 @@ def kosher_args(remote_ip, uri):
         return True not in all_false
 
 
+class EventNotFoundError(Exception):
+    """
+    Error when certain event cannot be found.
+    """
+
+    def __init__(self, details=''):
+        self.details = details
+        Exception.__init__(self)
+
+    def __str__(self):
+        return str(self.details)
+
+
+class EventTracker(object):
+
+    @staticmethod
+    def start_get_event(vm_name):
+        """
+        Use a virsh session with subcommand 'event' to catch events
+
+        :param vm_name: name of the vm to be catched
+        :return: the virsh session with 'event'
+        """
+        virsh_session = aexpect.ShellSession(VIRSH_EXEC)
+        event_cmd = 'event %s --all --loop' % vm_name
+        logging.info('Sending "%s" to virsh shell', event_cmd)
+        virsh_session.sendline(event_cmd)
+
+        return virsh_session
```

Comment:
I am a little why you return virsh_session when start_get_event, versus virsh_session as pass in parameter in finish_get_event.

Reply:
@chunfuwen that's because we need virsh session to get event

Reply:
Generally, when people see start_get_event() , the first impression occurred to people is that it need return True or False indicating whether event is started or not , rather than return one session. 
Moreover , start_get_event and finish_get_event are similar functions, but it make more consistently if both of them accept the same parameter ,e.g vm_name.

Reply:
Then how do you propose we should edit this function?

Reply:
The current functions are working well, I have tested. Feel free to suggest better implementations.

Reply:
Also, how necessary do we need to modify the function?

Reply:
Can we pass  virsh_session  as one parameter to those methods?

Reply:
Sorry that I just don't get it. How would you like to provide an example of what you want this function to be? I think it might be easier for me to understand your suggestion.

Reply:
Rough workflow follows as:

1)virsh_session = aexpect.ShellSession(VIRSH_EXEC)
2)start_get_event(virsh_session,xx)
3)finish_get_event(virsh_session,xx)

Reply:
After considering your proposal, I'd like to keep the function the way it is now, if it's not unacceptable to you.